### DISCUSSION TITLE
怎么给图片添加脚注呢？

### DISCUSSION THREAD

**Discussion by HongYegg** [sjtug/SJTUThesis#877] [提问]
![image](https://user-images.githubusercontent.com/62496073/209830515-2d7b01c0-ab7c-4615-83e3-ea5b8ad6a5f0.png)
我这样写了它只显示标号，用footnotemark和footnotetext的形式，图和文本不在一页。
\begin{figure}[!hbtp]
  \centering
  \subcaptionbox{谷歌自动驾驶汽车}%
                  {\includegraphics[width=0.4\textwidth]{figures/waymo.png}}
  \hspace{1cm}
  \subcaptionbox{特斯拉自动驾驶汽车}%
                  {\includegraphics[width=0.4\textwidth]{figures/tesla.jpg}}
  \hspace{1cm}
  \subcaptionbox{百度自动驾驶汽车}%
                  {\includegraphics[width=0.4\textwidth]{figures/baidu.png}}
  \hspace{1cm}
  \subcaptionbox{小马自动驾驶汽车}%
                  {\includegraphics[width=0.4\textwidth]{figures/pony.jpg}}
  \bicaption{不同厂商的自动驾驶汽车产品}
            {Self-driving car products from different manufacturers}
  \label{fig:self-driving car products}
  
  \footnote{hhhh}
  \footnote{\url{https://waymo.com/intl/zh-cn}}
  \footnote{\url{https://www.tesla.cn/drive}}
  \footnote{\url{https://www.apollo.auto}}
  \footnote{\url{https://www.pony.ai}}
  
\end{figure}

![image](https://user-images.githubusercontent.com/62496073/209830744-c4b12ebb-f037-4f6f-89c3-b8ba6225296d.png)

**Comment by LogCreative** [comment]
就我的了解而言，目前对于浮动体环境没有太好的做法，大概只能 `\footnotemark` 和 `\footnotetext` 分开，将 `\footnotetext` 写在想要出现页的文字位置。
```latex
\documentclass{sjtuthesis}
\usepackage{hyperref}
\begin{document}
    \begin{figure}
        \begin{subfigure}{0.45\textwidth}
            \caption{特斯拉自动驾驶汽车\protect\footnotemark}
        \end{subfigure}
    \end{figure}

    \footnotetext{\url{https://www.tesla.cn/drive}}
\end{document}
```
这个例子将脚注写在了 `\caption` 里，需要加 `\protect` 保护。

<img width="180" alt="截屏2022-12-29 10 55 53" src="https://user-images.githubusercontent.com/61653082/209897484-5148e910-158d-4255-bf81-a9c76b3e826b.png">

**Comment by LogCreative** [comment]
当然我更推荐的一种做法是将图片引用信息写在参考文献中。

```latex
\documentclass{sjtuthesis}
\usepackage[style=gb7714-2015]{biblatex}
\addbibresource{ref.bib}
\begin{document}

    \begin{filecontents}{ref.bib}
        @misc{tesla,
            title = {Tesla},
            url = {https://www.tesla.cn/drive},
        }
    \end{filecontents}

    \begin{figure}[h]
        \begin{subfigure}{0.45\textwidth}
            \caption{特斯拉自动驾驶汽车\cite{tesla}}
        \end{subfigure}
    \end{figure}

    \printbibliography
\end{document}
```
<img width="252" alt="截屏2022-12-29 11 04 45" src="https://user-images.githubusercontent.com/61653082/209898179-d810a769-d41d-46b1-8b29-c6be71b9a4f6.png">

**Comment by HongYegg** [comment]
谢谢


---原始邮件---
发件人: ***@***.***&gt;
发送时间: 2022年12月29日(周四) 上午10:58
收件人: ***@***.***&gt;;
抄送: ***@***.******@***.***&gt;;
主题: Re: [sjtug/SJTUThesis] 怎么给图片添加脚注呢？ (Discussion #877)




 
就我的了解而言，目前对于浮动体环境没有太好的做法，大概只能 \footnotemark 和 \footnotetext 分开，将 \footnotetext 写在想要出现页的文字位置。
 \documentclass{sjtuthesis} \usepackage{hyperref} \begin{document}     \begin{figure}         \begin{subfigure}{0.45\textwidth}             \caption{特斯拉自动驾驶汽车\protect\footnotemark}         \end{subfigure}     \end{figure}     \footnotetext{\url{https://www.tesla.cn/drive}} \end{document}
 
这个例子将脚注写在了 \caption 里，需要加 \protect 保护。
 

 
&mdash;
